### PR TITLE
[for v0.19]cmake should make a report after it is finished

### PR DESCRIPTION
**Pull request by mywave82**
Also includes various minor cmake updates.

### CODE REVIEW

**Comment by luzpaz** [comment]
Sorry to spam, but I'm very excited about this PR.

**Comment by mywave82** [comment]
It is a result of many trial and errors making FreeCAD compile, and I hope it will make things easier for others. Comments, critics and additions are very welcome.

**Comment by PrzemoF** [comment]
That might be really handy for users trying to compile FreeCAD for the first time on a  system. I had a problem with compilation on linux mint recently - I'll give it a shot again when it's merged.

**Comment by wwmayer** [comment]
I hadn't the time to look at this PR. But so close to the release of v0.18 I think it's better to postpone it to v0.19.

File: CMakeLists.txt [hunk 1]
```diff
@@ -573,8 +576,10 @@ endif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND NOT BUILD_WITH_CONDA)
     ENDIF(NOT PYTHONLIBS_FOUND)
 
 # -------------------------------- pcl ----------------------------------
-    #PCL needs to be found before boost because the PCLConfig also calls find_package(Boost ...),
-    #but with different components
+# Can be used by ReverseEngineering module"
```
Comment (luzpaz) [inline CMakeLists.txt]: It is also used by certain 3rd Party Addons
For example @microelly2 NURBS https://github.com/microelly2/freecad-nurbs

File: CMakeLists.txt [hunk 2]
```diff
@@ -573,8 +576,10 @@ endif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND NOT BUILD_WITH_CONDA)
     ENDIF(NOT PYTHONLIBS_FOUND)
 
 # -------------------------------- pcl ----------------------------------
```
Comment (luzpaz) [inline CMakeLists.txt]: I think we should have this instead
`# ------------------- pcl (pointclouds) -------------------------`

**Comment by wwmayer** [comment]
Merged.

**Pull request closed without merging**